src/Settings.cpp

```diff
@@ -596,13 +640,24 @@ void Settings::parseInput(int argc, char** argv)
             mCredsFile = optarg;
             break;
         case OPT_AUTHUSER:
+            // Need to manually check if we have our optional argument
+            if (optarg == NULL && optind < argc && argv[optind][0] != '-') {
+                optarg = argv[optind++];
+            }
```

Comment:
How is this supposed to work? If there's no optional argument it should take the next one, but not if it is a new flag? How is this possible, if there is no optional argument?

Reply:
It's related to this sort of issue: [getopt does not parse optional arguments to parameters](https://stackoverflow.com/questions/1052746/getopt-does-not-parse-optional-arguments-to-parameters)

--username and --password take optional parameters. If they're not supplied then the user gets prompted at the command line for them. The problem is that getopt_long wants long options to take the form --username=user when working out whether the next argument supplied is the optarg or whether it's the next command switch. So when users don't include the equals sign (as they typically don't) it treats 'user' as a command switch and then chews the user out for having a non-recognized command. This tests to see if the next string in argv should be interpreted as the optarg for these commands, or whether we should leave the parser to do its thing.

Reply:
(So this has been busted from the start. I must have just got lucky in the way I was originally testing it. (Only omitting one optional argument which I must have conveniently placed at the end of the command.)

Reply:
I don't understand your last comment, but with your explanation this sounds reasonable. Could you add a comment that this is about the case without the equal sign?

Reply:
Sure.

Reply:
Oh, I mean that this has never properly worked since the introduction of the code. I just got lucky in how I tested it. (It worked for one case, but not in general.)